hashes/Cargo.toml

```diff
@@ -15,8 +15,9 @@ exclude = ["tests", "contrib"]
 
 [features]
 default = ["std"]
-std = ["alloc", "hex/std", "bitcoin-io/std"]
-alloc = ["hex/alloc"]
+std = ["alloc", "hex/std", "bitcoin-io/std", "serde/std"]
+alloc = ["hex/alloc", "serde/alloc"]
```

Comment:
In b43344903a5aa7f470889fdb213a62a5f3c5f83d:

Innteresting. I remember it occurring to us that we could enable the `serde/alloc` feature, but since we didn't need it within the library, we didn't bother. But if we're depending on this within our tests, that's an indication that users probably need it too. So ACK making this change.

Reply:
It needs to be `serde?/std` and `serde?/alloc` respectively though.

Reply:
> It needs to be `serde?/std` and `serde?/alloc` respectively though.

I did not know this syntax existed, now looking at Cargo.toml again, shouldn't `"bitcoin-io/std"` be `"bitcoin-io?/std"` then?

Reply:
Maybe they should be, for prudence's sake. But `bitcoin-io` is not an optional dependency so it can never be turned off and the `?` syntax isn't needed.

Reply:
@antonilol it was pretty recent addition and we only bumped MSRV recently so it wasn't in this project for long. Also IIRC we're still missing some changes. It only applies to optional dependencies.

@apoelstra you can't use `?` in a non-optional dependency, so we can't blindly put it everywhere. But if someone doesn't want to bother, just add it everywhere and remove when cargo complains :)